zio-schema-bson/shared/src/main/scala/zio/bson/BsonCodecConfiguration.scala

```diff
@@ -0,0 +1,109 @@
+package zio.bson
+
+import java.util.regex.Pattern
+
+import zio.bson.BsonCodecConfiguration.SumTypeHandling.WrapperWithClassNameField
+import zio.bson.BsonCodecConfiguration._
+
+case class BsonCodecConfiguration(
+  sumTypeHandling: SumTypeHandling = WrapperWithClassNameField,
+  skipNullsInMap: Boolean = false,
+  skipNullsInCaseClass: Boolean = true,
+  classNameMapping: TermMapping = identity,
+  fieldNameMapping: TermMapping = identity,
+  allowExtraFields: Boolean = true
+)
+
+object BsonCodecConfiguration {
+  implicit val default: BsonCodecConfiguration = BsonCodecConfiguration()
+
+  type TermMapping = String => String
+
+  sealed trait SumTypeHandling
+
+  object SumTypeHandling {
+
+    /**
+     * Sum type hierarchy:
+     * {{{
+     *   sealed trait MySum
+     *   case class SomeBranch(a: Int)
```

Comment:
Are these case class examples supposed to extend the sealed trait?

Reply:
Thank you' I'll fix it.
Please note that it is unrelated to `zio-schema` derivation. It is for magnolia derivation. You can find `zio-bson` here: https://github.com/zivergetech/zio-mongo
`aio,bson` package should not be in this MR - I'll publish `zio-bson` today.